src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs

```diff
@@ -0,0 +1,479 @@
+                var handler = new TcpClientHandler(this, Logging.GetLogger(System, typeof(TcpClientHandler)), remoteAddress);
+                pipeline.AddLast("ClientHandler", handler);
+            }
+        }
+
+        private void SetServerPipeline(IChannel channel)
+        {
+            if (Settings.EnableSsl)
+            {
+                channel.Pipeline.AddFirst("TlsHandler", TlsHandler.Server(Settings.Ssl.Certificate));
+            }
+
+            SetInitialChannelPipeline(channel);
+            var pipeline = channel.Pipeline;
+
+            if (Settings.TransportMode == TransportMode.Tcp)
+            {
+                var handler = new TcpServerHandler(this, Logging.GetLogger(System, typeof(TcpServerHandler)), AssociationListenerPromise.Task);
+                pipeline.AddLast("ServerHandler", handler);
+            }
+        }
+
+        private ServerBootstrap ServerFactory()
+        {
+            if (InternalTransport != TransportMode.Tcp)
+                throw new NotSupportedException("Currently DotNetty server supports only TCP tranport mode.");
+
+            var addressFamily = Settings.DnsUseIpv6 ? AddressFamily.InterNetworkV6 : AddressFamily.InterNetwork;
+
+            var server = new ServerBootstrap()
+                .Group(serverEventLoopGroup)
+                .Option(ChannelOption.SoReuseaddr, Settings.TcpReuseAddr)
+                .Option(ChannelOption.SoKeepalive, Settings.TcpKeepAlive)
+                .Option(ChannelOption.TcpNodelay, Settings.TcpNoDelay)
+                .Option(ChannelOption.AutoRead, false)
+                .Option(ChannelOption.SoBacklog, Settings.Backlog)
+                .ChannelFactory(() => Settings.EnforceIpFamily
+                    ? new TcpServerSocketChannel(addressFamily)
+                    : new TcpServerSocketChannel())
+                .ChildHandler(new ActionChannelInitializer<TcpSocketChannel>(SetServerPipeline));
+
+            if (Settings.ReceiveBufferSize.HasValue) server.Option(ChannelOption.SoRcvbuf, Settings.ReceiveBufferSize.Value);
+            if (Settings.SendBufferSize.HasValue) server.Option(ChannelOption.SoSndbuf, Settings.SendBufferSize.Value);
+            if (Settings.WriteBufferHighWaterMark.HasValue) server.Option(ChannelOption.WriteBufferHighWaterMark, Settings.WriteBufferHighWaterMark.Value);
+            if (Settings.WriteBufferLowWaterMark.HasValue) server.Option(ChannelOption.WriteBufferLowWaterMark, Settings.WriteBufferLowWaterMark.Value);
+
+            return server;
+        }
+
+        private async Task<IPEndPoint> ResolveNameAsync(DnsEndPoint address)
+        {
+            var resolved = await Dns.GetHostEntryAsync(address.Host);
+            //NOTE: for some reason while Helios takes first element from resolved address list
+            // on the DotNetty side we need to take the last one in order to be compatible
+            return new IPEndPoint(resolved.AddressList[resolved.AddressList.Length - 1], address.Port);
```

Comment:
It depends on `enforce-ip-family` setting. It IP address family is on, second option will be used. I also think, we should pick first correct address, this however was causing a failure between Helios and DotNetty backward compatibility spec - I don't know yet why.